src/vtl.ts

```diff
@@ -261,9 +264,17 @@ export abstract class VTL {
     } else if (isBreakStmt(node)) {
       return this.add("#break");
     } else if (isCallExpr(node)) {
-      const serviceCall = findIntegration(node);
-      if (serviceCall) {
-        return this.integrate(serviceCall, node);
+      if (isReferenceExpr(node.expr)) {
+        const ref = node.expr.ref();
+        if (isIntegration<Integration>(ref)) {
+          const serviceCall = new IntegrationImpl(ref);
+          return this.integrate(serviceCall, node);
+        } else {
+          throw new SynthError(
+            ErrorCodes.Unexpected_Error,
+            "Called references are expected to be an integration."
```

Comment:
Is this error code too general?

Reply:
I don't think a user should be able to create this right now. We can change it if that turns out to be wrong.